File: languages/go/entity/identityV2/credential_status.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// CredentialStatus represents the status information for a credential, including its ID, revocation nonce, and type.
type CredentialStatus struct {
	Id              string
	RevocationNonce int64
	Type            string
}

func NewCredentialStatusFromProto(s *proto.CredentialStatusV2) CredentialStatus {
	if s == nil {
		return CredentialStatus{}
	}
	return CredentialStatus{
		Id:              s.Id,
		RevocationNonce: s.RevocationNonce,
		Type:            s.Type,
	}
}

func (c CredentialStatus) ToProto() *proto.CredentialStatusV2 {
	return &proto.CredentialStatusV2{
		Id:              c.Id,
		RevocationNonce: c.RevocationNonce,
		Type:            c.Type,
	}
}

File: languages/python/bloock/entity/identity_v2/credential_status.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto


class CredentialStatus:
    def __init__(self, id: str, revocation_nonce: int, type: str) -> None:
        self.id = id
        self.revocation_nonce = revocation_nonce
        self.type = type

    @staticmethod
    def from_proto(c: proto.CredentialStatusV2) -> CredentialStatus:
        return CredentialStatus(
            id=c.id,
            revocation_nonce=c.revocation_nonce,
            type=c.type
        )

    def to_proto(self) -> proto.CredentialStatusV2:
        return proto.CredentialStatusV2(
            id=self.id,
            revocation_nonce=self.revocation_nonce,
            type=self.type
        )
